src/main/java/com/sap/lsp/cf/ws/LanguageServerFilter.java

```diff
@@ -26,7 +26,7 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
         String requestUuid = extractDIToken(servletRequest);
         if (System.getenv().containsKey(DI_TOKEN_ENV)) {
             String diToken = System.getenv(DI_TOKEN_ENV);
-            if (!diToken.equals(requestUuid)) {
+            if (diToken == null || !diToken.equals(requestUuid)) {
                 LOG.warning("Invalid request - token mismatch. Got token " + requestUuid);

```

Comment:
did you get any warning about printing requestUuid to log?

Reply:
no the standard want us to check null pointer exceptions in this case